src/test/java/be/seeseemelk/mockbukkit/entity/EntityMockTest.java

```diff
@@ -441,4 +441,24 @@ void setGliding()
 		assertFalse(player.isGliding());
 	}
 
+	@Test
+	void testHasAI()
+	{
+		LivingEntity zombie = (LivingEntity) world.spawnEntity(new Location(world, 10, 10, 10), EntityType.ZOMBIE);
+
+		assertTrue(zombie.hasAI());
+	}
+
+	@Test
+	void testSetAI()
+	{
+		LivingEntity zombie = (LivingEntity) world.spawnEntity(new Location(world, 10, 10, 10), EntityType.ZOMBIE);
+
+		zombie.setAI(false);
+		assertFalse(zombie.hasAI());
+
+		zombie.setAI(true);
+		assertTrue(zombie.hasAI());
+	}
+
```

Comment:
There should be a test to ensure non-mob entities have no ai, and that setting it doesn't change that.